packages/azure/data_stream/activitylogs/elasticsearch/ingest_pipeline/default.yml

```
@@ -313,6 +313,23 @@ processors:
   - set:
       field: event.kind
       value: event
+  - script:
+      description: Appends principal_id and resource_id to `related.entity`
+      lang: painless
+      ignore_failure: true
+      on_failure: 
+        - set:
+            description: Add error reason
+            field: error.message
+            value: "{{{ _ingest.on_failure_message }}}"
+      source: |
+        ctx.related = ctx.related ?: [:];
+        ctx.related.entity = new HashSet();
+        ctx.related.entity.add(field("azure.activitylogs.identity.authorization.evidence.principal_id").get(null));
+        ctx.related.entity.add(field("azure.resource_id").get(null));
+        ctx.related.entity.remove("");
+        ctx.related.entity.remove(null);
+
```

Suggest rather the simpler
```suggestion
  - append:
      field: related.entity
      value: '{{{azure.resource_id}}}'
      allow_duplicates: false
      if: >
        ctx.azure?.resource_id != null && ctx.azure.resource_id != ''
  - append:
      field: related.entity
      value: '{{{azure.activitylogs.identity.authorization.evidence.principal_id}}}'
      allow_duplicates: false
      if: >
        ctx.azure?.activitylogs?.identity?.authorization?.evidence?.principal_id != null &&
        ctx.azure.activitylogs.identity.authorization.evidence.principal_id != ''
```
which is shorter, does not require an error handler, and does not allocate in the case that the fields are absent.

---

i did this, but currently if one of these append processors fails, the pipeline won't continue, which is something i'd like to avoid. i think adding an empty `on_failure` field to the processors should cover that, but i didn't manage to test this with invalid values. wdyt? how could i ensure a failure won't stop the pipeline from executing the next steps, but still log the error? (`ignore_failure` is also not preferable)

---

What is the failure mode that could lead to an append failing?

---

tbh i don't know. i don't see how it can fail now, but as a general good measure i assumed it'd be better to account for it in anyway. but if you're ok with it then i guess it's fine.

---

I'm OK with it.